packages/cozy-client/src/hooks/useQuery.spec.jsx

```diff
@@ -159,6 +166,29 @@ describe('use query', () => {
     expect(lastCallArgs).toMatchObject({ skip: current.data.length })
     expect(fetchMoreResult).toEqual({ data: null })
   })
+
+  it('should call onError callback when an error is thrown. Callback passed as argument', async () => {
+    jest.spyOn(console, 'error').mockImplementation(() => null)
+    const requestHandler = jest.fn()
+    const client = new CozyClient({
+      links: [new CozyLink(requestHandler)]
+    })
+    const error = new TypeError('Failed to fetch')
+    requestHandler.mockReturnValue(Promise.reject(error))
+    const onError = jest.fn().mockReturnValue(() => true)
+    setupQuery({
+      customClient: client,
+      queryOptions: {
+        as: 'fake-query',
+        onError
+      },
+      queryDefinition: () => Q('io.cozy.todos')
+    })
+
+    setTimeout(() => {
+      expect(onError).toBeCalled()
+    }, 100)
```

Comment:
This is alway a bad idea to rely on timeout. 

```
it('should query through the client in context', () => {
    const { client } = setupQuery({
      queryOptions: {
        as: 'simpsons',
        onError
      },
      queryDefinition: () => Q('io.cozy.simpsons')
    })
    expect(onError).toHaveBeenCalled()
  })
```

this doesn't work?

Reply:
I know, but it doesn't work any other way